owncloudApp/src/main/java/com/owncloud/android/ui/adapter/FileListListAdapter.java

```diff
@@ -413,6 +417,18 @@ public void swapDirectory(OCFile folder, FileDataStorageManager updatedStorageMa
 
         mFiles = FileStorageUtils.sortFolder(mFiles, FileStorageUtils.mSortOrderFileDisp,
                 FileStorageUtils.mSortAscendingFileDisp);
+
+        final SharedPreferences prefs = android.preference.PreferenceManager.getDefaultSharedPreferences(mContext);
+        boolean showHiddenFiles = prefs.getBoolean(SHOW_HIDDEN_FILES, true);
+
+        if (!showHiddenFiles) {
+            for (int i = 0; i < mFiles.size(); i++) {
+                if (mFiles.get(i).getFileName().startsWith(".")) {
+                    mFiles.remove(i);
+                    i--; // Otherwise it will skip the element after the removed index
+                }
+            }
+        }
```

Comment:
BTW, I would extract this into another function to filter the list

Reply:
done